src/shogun/distributions/kernel_exp_family/impl/Base.cpp

```diff
@@ -112,20 +112,26 @@ SGVector<float64_t> Base::log_pdf(const SGMatrix<float64_t> X)
 	SGVector<float64_t> result(N_test);
 #pragma omp parallel for
 	for (auto i=0; i<N_test; ++i)
-		result[i] = log_pdf(i);
+		result[i] = ((const Base*)this)->log_pdf(i);
 
 	return result;
 }
 
 float64_t Base::log_pdf(SGVector<float64_t> x)
 {
 	set_test_data(x);
-	return log_pdf(0);
+	return ((const Base*)this)->log_pdf(0);
```

Comment:
Okok

On Thursday, 7 July 2016, Esben Sørig <notifications@github.com> wrote:

> In src/shogun/distributions/kernel_exp_family/impl/Base.cpp
> <https://github.com/shogun-toolbox/shogun/pull/3355#discussion_r69956349>:
>
> >
> >  	return result;
> >  }
> >
> >  float64_t Base::log_pdf(SGVector<float64_t> x)
> >  {
> >  	set_test_data(x);
> > -	return log_pdf(0);
> > +	return ((const Base*)this)->log_pdf(0);
>
> Fixes the ambiguity warnings. On my Mac it didn't compile because of this
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/shogun-toolbox/shogun/pull/3355/files/b45bdc86ebc800e8c2cfbf1a6268b91fee09bc9b#r69956349>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAqqvy2iJLqpsG39Zb7mTNYY38nzZj44ks5qTUA0gaJpZM4JG80d>
> .
>


-- 
Sent from my phone